tests/test_ndimage.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2019 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Modifications made by Tim Mensinger, 2024.
+
+from functools import partial
+
+import jax.numpy as jnp
+import jax.scipy.ndimage
+import numpy as np
+import pytest
+import scipy.ndimage
+from numpy.testing import assert_allclose, assert_array_almost_equal, assert_array_equal
+
+import lcm.ndimage
+
+jax_map_coordinates = partial(jax.scipy.ndimage.map_coordinates, order=1, cval=0)
+scipy_map_coordinates = partial(scipy.ndimage.map_coordinates, order=1, cval=0)
+lcm_map_coordinates = lcm.ndimage.map_coordinates
+
+JAX_IMPLEMENTATIONS = [jax_map_coordinates, lcm_map_coordinates]
+
+
+TEST_SHAPES = [
+    (5,),
+    (3, 4),
+    (3, 4, 5),
+]
+
+TEST_COORDINATES_SHAPES = [
+    (7,),
+    (2, 3, 4),
+]
+
+
+def _make_test_data(shape, coordinates_shape, dtype):
+    rng = np.random.default_rng()
+    x = np.arange(np.prod(shape), dtype=dtype).reshape(shape)
+    c = [(size - 1) * rng.random(coordinates_shape).astype(dtype) for size in shape]
+    return x, c
+
+
+@pytest.mark.parametrize("map_coordinates", JAX_IMPLEMENTATIONS)
+@pytest.mark.parametrize("shape", TEST_SHAPES)
+@pytest.mark.parametrize("coordinates_shape", TEST_COORDINATES_SHAPES)
+@pytest.mark.parametrize("dtype", [np.int64, np.float64])
+def test_map_coordinates_against_scipy(
+    map_coordinates, shape, coordinates_shape, dtype
+):
+    """Test that all libraries implement same behavior with integer input."""
+    x, c = _make_test_data(shape, coordinates_shape, dtype=dtype)
+
+    x_jax = jnp.asarray(x)
+    c_jax = [jnp.asarray(c_i) for c_i in c]
+
+    expected = scipy_map_coordinates(x, c)
+    got = map_coordinates(x_jax, c_jax)
+
+    assert_array_almost_equal(got, expected, decimal=14)
+
+
+@pytest.mark.parametrize("map_coordinates", JAX_IMPLEMENTATIONS)
+def test_map_coordinates_round_half_integer_input(map_coordinates):
```

Comment:
Just combine with below test and add another parametrization. Usually we try to avoid this, but the test body is complex, so repeating it is worse than additionally looping over dtypes.